### ISSUE TITLE
`No default application for selecting a file` even though there is one.

### ISSUE THREAD

**Issue by lubosz** [cdskev/ginlo-android#8]
When trying to open a file from ginlo, the application reports `No default application for selecting a file`, even though a default application for the file type is installed.

The error is the following:
```
No default application for selecting a file of type application/pdf was found. You could alternatively try to share the file.
```
```
No default application for selecting a file of type application/vnd.openxmlformats-officedocument.wordprocessingml.document was found. You could alternatively try to share the file.
```
```
No default application for selecting a file of type application/msword was found. You could alternatively try to share the file.`
```

In German:
```
Es wurde keine Standardanwendung für die Auswahl einer Datei des Typs application/pdf gefunden. Sie können alternativ versuchen, die Datei zu teilen.
```

I was able to encounter this error for `pdf`, `doc` and `docx` formats.
Interestingly the error disappears for `pdf`, where I am able to share the file to Google Drive and opening it with Google Drive. After these steps I am able to open `pdf` files through the `open` command in ginlo, even new pdf files.

These steps are not reproducible for `doc` and `docx` formats, where the `No default application` error persists, even when following the steps that worked for `pdf`. So for every `doc` or `docx` I need to first share it with a generic file capable application, upload it, download it and only then I am able to open such files from the `Files` application.

I am able to download a `doc` or `docx` file and get correct suggested default applications form the `Files` app.

In both cases default applications were installed and opened before ginlo.

These issues could be reproduced on 2 separate Android 12 devices.

To summarize:
* ginlo does not find default applications for `pdf` and `doc/docx` files
* There are default applications installed and these files can be opened with the system `File` app.
* In the case of `pdf` a persistent workaround exists by triggering a sequence through share.
* Telegram is able to open correct indents for both `pdf` and `doc/docx`.

Even though this workaround is kind of hinted in a UI popup, this is suboptimal user experience and users like my wife had trouble of getting it to work at all. In addition the workaround does not work for doc.

**Comment by karstenschramm** [comment]
We have been working on that, coming to the result that this is not a problem of ginlo. Many other apps do not use 'open' but 'share' functionality to allow for handling 3rd party files. That often works - and is possible for our ginlo-users, too.

New Android security guidelines became mandatory by the end of last year. All apps that have not been refactored to meet the new requirements are not being offered to the users when opening files.

At least with the often used pdf-format, we found a solution: ginlo is now equipped with a built-in reader that may be - of course -  switched off in settings. In other cases the use of 'share' instead of 'open' may help.

**Comment by lubosz** [comment]
> this is not a problem of ginlo

Yes this is exactly a problem of ginlo. It works like a charm in all decent messengers like Telegram and WhatsApp.
This is just an excuse to not study the indent sharing mechanism on Android and provide a decent implementation.
Do something like this:
https://stackoverflow.com/questions/31621419/android-how-do-i-open-a-file-in-another-app-via-intent

Also put the file some place other apps can read it. In your case you also need to decrypt it first.

> Many other apps do not use 'open' but 'share' functionality to allow for handling 3rd party files.

This is a catastrophical UX. sharing != opening. 
3 months after you closed this issue I still need to *share* files with myself on Telegram (which has implemented this correctly) to *open* them in the correct app that is available on my phone.

So sharing != opening. These are different things on Andoid.
If I share a file to Google Docs it want to upload to a folder. Opening it is still 2 more extra steps (finding it in the app and opening it)
While when I open it, it just opens.

> At least with the often used pdf-format, we found a solution

A build in reader is a horrible and hard to maintain workaround. I would never call that a solution.
You can't have a build in reader for all file formats. I wanted to open doc today. Telegram was able to do it, without a buildin reader.
Just implement the launcher indent correctly for the correct mime types and put the decyphered file in a place other apps can read it.

**Comment by lubosz** [comment]
This is the implementation in Telegram:
https://github.com/TelegramOrg/Telegram-Android/blob/master/TMessagesProj/src/main/java/org/telegram/messenger/AndroidUtilities.java#L3014

**Comment by kschramm-ginlo** [comment]
We have been working on that. Issue will be solved in v5.4.7
